### ISSUE TITLE
Can helm chart requirements have templated overrides?

### ISSUE THREAD

**Issue by mhamilton723** [helm/helm#4822]
More specifically I am trying to understand if I could do the following:

Chart A Values.yaml:
```
foo: "here"
```
Chart B requirements.yaml:
```
dependencies:
  - name: a
    repository: ...
    version: ...
```
Chart B values.yaml (or somewhere in the parent chart):
```
a.foo = "{{.Values.someKeyFromB}} nontrivial template text {{.Values.someOtherKeyFromB}}"
```

Basically I would like to do something like overriding a field with something more complex. Thanks in advance!

**Comment by bacongobbler** [comment]
Hey @mhamilton723, for the same reasons stated in #2492 - which asks for templating in values.yaml - we likely will not support templates in requirements.yaml. Please have a look at @thomastaylor312's comment on the more in-depth reason on why we don't support templates outside of the core rendering engine: https://github.com/helm/helm/issues/2492#issuecomment-413635332

That being said, the Lua template engine being introduced in Helm 3 may prove useful for your use case in the same way it may prove useful for #2492. Once we have an alpha release of Helm 3 ready with the Lua template engine available, we'd love to hear your feedback.

Thanks!

**Comment by mhamilton723** [comment]
@bacongobbler thanks for the reply. To clarify, I don’t actually need to template the requirements.yaml. I need something that will allow me to set the variables of a child chart with templates values. This will in effect allow me to compose charts in a way that the API of the composition can be simpler than just both of the APIs combined together with name spacing. Does that make sense and is their any way to hack this kind of behavior?

**Comment by bacongobbler** [comment]
Oh I see.

If you want to override subchart values from an "umbrella chart", something like this should work:

```
a:
  foo: "someothervalue"
```

We used this quite often in the Deis Workflow chart way back: https://github.com/deis/workflow/blob/889f6ce1633dddc52c84eba168f76f57d18bab3e/charts/workflow/values.yaml#L126-L136

As far as more complex examples like the one you're showing, that likely will not work because you're essentially asking the template engine to render values as templates, which we decided not to support in Helm 2 for the reasons listed in that PR.

Hope that helps!
